### PR TITLE
raftstore: add some test cases for the CPU-based Load Base Split strategy

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What is changed and how it works?

Issue Number: ref #12063.

What's Changed:

```commit-message
Add some test cases for the CPU-based Load Base Split strategy.
```

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
None.
```

### CODE REVIEW

**Review by AndreMouche — APPROVED** [review]
LGTM

File: components/raftstore/src/store/worker/split_controller.rs [hunk 1]
```diff
@@ -1563,6 +1681,96 @@ mod tests {
...
+    fn test_collect_cpu_stats() {
+        let auto_split_controller = AutoSplitController::default();
+        let region_cpu_map = auto_split_controller.collect_cpu_stats(vec![]);
+        assert!(region_cpu_map.is_empty());
+
+        let ab_key_range_tag = Arc::new(TagInfos {
+            store_id: 0,
+            region_id: 1,
+            peer_id: 0,
+            key_ranges: vec![(b"a".to_vec(), b"b".to_vec())],
```
Comment (nolouch) [inline components/raftstore/src/store/worker/split_controller.rs:1694]: could you add a test case about one tag which has multi ranges? as you said, it exists.

Comment (JmPotato) [reply components/raftstore/src/store/worker/split_controller.rs:1694]: Ok, I will add later.

Comment (JmPotato) [reply components/raftstore/src/store/worker/split_controller.rs:1694]: Added. PTAL.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**